### PR TITLE
Adds support for multi-value members in AST analysis.

### PR DESCRIPTION
**Pull request by zooba**

### CODE REVIEW

**Comment by zooba** [comment]
Still need to do some specific unit tests, but the core functionality is there.

File: Python/Product/Analysis/Interpreter/Ast/NameLookupContext.cs [hunk 1]
```diff
@@ -202,6 +203,23 @@ bool includeLocationInfo
             return null;
         }
 
+        public IEnumerable<IPythonType> GetTypesFromValue(IMember value) {
+            if (value is IPythonMultipleMembers mm) {
+                var seen = new HashSet<IPythonType>();
```
Comment (int19h) [inline Python/Product/Analysis/Interpreter/Ast/NameLookupContext.cs]: This looks like `Enumerable.Select(...).Distinct()`.

File: Python/Product/Analysis/Interpreter/Ast/AstPythonMultipleMembers.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                return new AstPythonMultipleMembers(new[] { x, y });
+            }
+        }
+
+        public void AddMember(IMember member) {
+            var old = Members;
+            if (!old.Contains(member)) {
+                Members = old.Concat(Enumerable.Repeat(member, 1)).ToArray();
+            } else {
+                Members = new[] { member };
```
Comment (huguesv) [inline Python/Product/Analysis/Interpreter/Ast/AstPythonMultipleMembers.cs]: You are also removing any previous members here, which seem wrong.

Comment (zooba) [reply Python/Product/Analysis/Interpreter/Ast/AstPythonMultipleMembers.cs]: Ah, you're right... made the simplification and didn't think about the changed else case...

Comment (zooba) [reply Python/Product/Analysis/Interpreter/Ast/AstPythonMultipleMembers.cs]: See #3119

**Pull request merged**